.github/workflows/test-kind.yml

```diff
@@ -6,15 +6,20 @@ jobs:
   build-and-test:
     runs-on: ubuntu-latest
     steps:
-      - name: Install GoLang
+      - name: "Install GoLang"
         uses: actions/setup-go@v2
         with:
           go-version: 1.16.x
-      - name: Checkout Repo
+      - name: "Checkout Repo"
         uses: actions/checkout@v2
-      - name: Build CLI
+      - name: "Create k8s Kind Cluster"
+        uses: helm/kind-action@v1.2.0
+      - name: "Build CLI"
         run: make build-cli-linux
-      - name: Make Packages
+      - name: "Make Packages"
         run: make init-package package-example-game package-example-data-injection package-example-gitops-data package-example-compose
-      - name: Run Tests
-        run: TESTDISTRO=kind make test-e2e
+      - name: "Run Tests"
+        # NOTE: We want to test providing a cluster to the test framework so this one creates its own KinD cluster
+        #       rather than having the test suite do it. The K3d tests do a self-provisioned cluster and the K3s tests
```

Comment:
Is there a particular reason to have the test suite create a k3d cluster vs do what we are doing for kind?  I have no preference, just the inconsistency is confusing.

Reply:
We wanted to make sure the test pipeline tested `TESTDISTRO=provided`. That's the only reason one creates the cluster and one expects one to exist.